src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -198,59 +203,59 @@ public Result listSubscriptions(@Nullable final String owningApplication, @Nulla
         } catch (final NoSuchSubscriptionException e) {
             LOG.debug("Failed to find subscription: {}", subscriptionId, e);
             return Result.problem(e.asProblem());
-        } catch (final ServiceUnavailableException e) {
+        } catch (final ServiceTemporarilyUnavailableException e) {
             LOG.error("Error occurred when trying to delete subscription: {}", subscriptionId, e);
-            return Result.problem(e.asProblem());
+            return Result.problem(Problem.valueOf(Response.Status.SERVICE_UNAVAILABLE, e.getMessage()));
         } catch (final NoSuchEventTypeException | InternalNakadiException e) {
             LOG.error("Exception can not occur", e);
             return Result.problem(e.asProblem());
         }
     }
 
-    public ItemsWrapper<SubscriptionEventTypeStats> getSubscriptionStat(final String subscriptionId)
+    public ItemsWrapper<SubscriptionEventTypeStats> getSubscriptionStat(final String subscriptionId,
+                                                                        final boolean includeDistance)
             throws InconsistentStateException, NoSuchSubscriptionException, ServiceTemporarilyUnavailableException {
         final Subscription subscription;
         try {
             subscription = subscriptionRepository.getSubscription(subscriptionId);
-        } catch (final ServiceUnavailableException ex) {
+        } catch (final ServiceTemporarilyUnavailableException ex) {
             throw new InconsistentStateException(ex.getMessage());
         }
-        final List<SubscriptionEventTypeStats> subscriptionStat = createSubscriptionStat(subscription);
+        final List<SubscriptionEventTypeStats> subscriptionStat = createSubscriptionStat(subscription, includeDistance);
         return new ItemsWrapper<>(subscriptionStat);
     }
 
-
-    private List<SubscriptionEventTypeStats> createSubscriptionStat(final Subscription subscription)
+    private List<SubscriptionEventTypeStats> createSubscriptionStat(final Subscription subscription,
+                                                                    final boolean includeDistance)
             throws InconsistentStateException, ServiceTemporarilyUnavailableException {
+        final List<EventType> eventTypes = getEventTypesForSubscription(subscription);
+        final List<PartitionEndStatistics> topicPartitions = loadPartitionEndStatistics(eventTypes);
```

Comment:
As I see we do this call not looking at `includeDistance` value. But this method will actually go to Kafka and do this expensive `seek` operation. I thought that in case of light-stats we will not check the offsets in Kafka.

Reply:
That is correct, I thought of doing it then forgot to do so. Looks like there will be some more refactoring.

Reply:
@v-stepanov 9436dff